monitoring/metrics.rb

```diff
@@ -0,0 +1,249 @@
+    group_by_fields: ["resource.zone"]
+  )
+
+  results = client.list_time_series(
+    project_name,
+    'metric.type = "compute.googleapis.com/instance/cpu/utilization"',
+    interval,
+    Google::Monitoring::V3::ListTimeSeriesRequest::TimeSeriesView::FULL,
+    aggregation: aggregation
+  )
+  results.each do |result|
+    p result
+  end
+
+  # [END monitoring_read_timeseries_reduce]
+end
+
+def list_metric_descriptors project_id:
+  # [START monitoring_list_descriptors]
+  client = Google::Cloud::Monitoring::Metric.new
+  project_name = Google::Cloud::Monitoring::V3::MetricServiceClient.project_path(project_id)
+  results = client.list_metric_descriptors(project_name)
+  results.each do |descriptor|
+    p descriptor.type
+  end
+  # [END monitoring_list_descriptors]
+end
+
+def list_monitored_resources project_id:
+  # [START monitoring_list_resources]
+  client = Google::Cloud::Monitoring::Metric.new
+  project_name = Google::Cloud::Monitoring::V3::MetricServiceClient.project_path(project_id)
+  results = client.list_monitored_resource_descriptors(project_name)
+  results.each do |descriptor|
+    p descriptor.type
+  end
+  # [END monitoring_list_resources]
+end
+
+def get_monitored_resource_descriptor project_id:, resource_type:
+  # [START monitoring_get_resource]
+  client = Google::Cloud::Monitoring::Metric.new
+  resource_path = Google::Cloud::Monitoring::V3::MetricServiceClient.monitored_resource_descriptor_path(
+    project_id,
+    resource_type
+  )
+
+ result = client.get_monitored_resource_descriptor(resource_path)
+ p result
+
+ # [END monitoring_get_resource]
+end
+
+def get_metric_descriptor metric_name:
+  # [START monitoring_get_descriptor]
+  client = Google::Cloud::Monitoring::Metric.new
+  descriptor = client.get_metric_descriptor(metric_name)
+  p descriptor
+  # [END monitoring_get_descriptor]
+end
+
+if __FILE__ == $PROGRAM_NAME
+  case ARGV.shift
+  when "create_metric_descriptor"
+    create_metric_descriptor project_id: ARGV.shift
+  when "delete_metric_descriptor"
+    delete_metric_descriptor descriptor_name: ARGV.shift
+  when "write_time_series"
+    write_time_series project_id: ARGV.shift
+  when "list_time_series"
+    list_time_series project_id: ARGV.shift
+  when "list_time_series_header"
+    list_time_series_header project_id: ARGV.shift
+  when "list_time_series_aggregate"
+    list_time_series_aggregate project_id: ARGV.shift
+  when "list_time_series_reduce"
+    list_time_series_reduce project_id: ARGV.shift
+  when "list_metric_descriptors"
+    list_metric_descriptors project_id: ARGV.shift
+  when "list_monitored_resources"
+    list_monitored_resources project_id: ARGV.shift
+  when "get_monitored_resource_descriptor"
+    get_monitored_resource_descriptor project_id: ARGV.shift, resource_type: ARGV.shift
+  when "get_metric_descriptor"
+    get_metric_descriptor metric_name: ARGV.shift
+  else
+    puts <<-usage
```

Comment:
Please use all caps for the string delimiter, e.g. `USAGE`.

Also, the library requires Ruby 2.3 or later, so you can use a [squiggly heredoc](https://infinum.co/the-capsized-eight/multiline-strings-ruby-2-3-0-the-squiggly-heredoc) which will make the indentation look better.

Reply:
Fixed